crates/stc_ts_file_analyzer/src/analyzer/assign/tpl.rs

```diff
@@ -27,6 +27,7 @@ impl Analyzer<'_, '_> {
     /// orders.
     ///
     /// After splitting, we can check if each element is assignable.
+    #[allow(clippy::needless_pass_by_ref_mut)]
```

Comment:
Complaining about `data` not being mutably used here (though it's not even actually used, there's an `allow(unused)`)

I didn't want to change APIs here, but that's probably more correct.